dist/index.js

```diff
@@ -97,6 +97,15 @@ async function installSamCli(python, version) {
   if (!version) {
     version = "1.*";
   }
+  // NOTE: cython 3.0.0 breaking installation of pyyaml - https://github.com/yaml/pyyaml/issues/724
+  await exec.exec(pythonPath, [
+    "-m",
+    "pip",
+    "install",
+    "cython<3.0.0",
+    "pyyaml==5.4.1",
```

Comment:
How long has it pinned to this? Thinking whether there's any meaningful impact to customers potentially pinning an older version of SAM CLI.

Reply:
it's been quite a while. The "serverlessrepo" dependency pinned `pyyaml` to `~=5.1` and the repo is read-only. So for now we are unable to bump PyYAML version.

IMO pinning to old version of SAM CLI wouldn't help the problem, because the error occurs while it's running `pip install aws-sam-cli` with the latest version of Cython

Reply:
Afaik: since cfn-lint was taken on as a dep: https://github.com/aws/aws-sam-cli/commit/376c394e09bb4a019ebb907a750573094f336a17

However, the latest release is now v6 on pyyaml.